src/ripple/app/misc/AmendmentTableImpl.cpp

```diff
@@ -97,51 +101,110 @@ class AmendmentTableImpl : public AmendmentTable
     amendmentList_t getDesired();    // amendments we support, do not veto, are not enabled
 };
 
-void
-AmendmentTableImpl::addInitial ()
+namespace detail
 {
-    // For each amendment this version supports, construct the AmendmentState object by calling
-    // addKnown. Set any vetoes or defaults. A pointer to the AmendmentState can be stashed
+using PreEnabledAmendmentsCollection = std::vector<AmendmentName>;
+
+/** preEnabledAmendments is a static collection of amendments are are enabled at build time.
+
+   Add amendments to this collection at build time to enable them on this server.
+*/
+
+PreEnabledAmendmentsCollection const preEnabledAmendments;
 }
 
-AmendmentState*
-AmendmentTableImpl::getCreate (uint256 const& amendmentHash, bool create)
+void
+AmendmentTableImpl::addInitial (Section const& section)
 {
-    // call with the mutex held
-    auto iter (m_amendmentMap.find (amendmentHash));
-
-    if (iter == m_amendmentMap.end())
+    for (auto const& a : detail::preEnabledAmendments)
     {
-        if (!create)
-            return nullptr;
+        if (!a.valid ())
+        {
+            std::string const errorMsg =
+                (boost::format (
+                     "preEnabledAmendments contains an invalid hash (expected "
+                     "a hex number). Value was: %1%") %
+                 a.hexString ()).str ();
+            throw std::runtime_error (errorMsg);
+        }
+    }
 
-        AmendmentState* amendment = & (m_amendmentMap[amendmentHash]);
+    detail::PreEnabledAmendmentsCollection toAdd (detail::preEnabledAmendments);
 
+    {
+        // add the amendments from the config file
```

Comment:
Less than a nit: My friend, Mr. Swirly, has berated me for non-programming text (e.g., comments) that doesn't have an initial cap and a closing period.  I think he has a point about improved readability.  I've been monitoring my own comments so I'm less likely to get hammered if he reviews my code.  You might consider the same habit.